src/preact/hooks/useDeepValue.ts

```diff
@@ -0,0 +1,68 @@
+import {useRef} from '#preact';
+
+export type Comparer<T> = (a: T, b: T) => boolean;
+
+/**
+ * Deep-compares the value against the previous value.
+ * If they're the same, the previous value is returned.
+ * Especially useful for dependency arrays.
+ *
+ * @param value - Any object, array, or value, which will be deep-compared
+ * @param compare - Should return `true` if the values are the same
+ */
+export function useDeepValue<TObj>(
+  value: TObj,
+  compare: Comparer<TObj> = fastDeepEqual
+): TObj {
+  const previousValue = useRef(value);
+  if (!compare(previousValue.current, value)) {
+    previousValue.current = value;
+  }
+  return previousValue.current;
+}
+
+// Modified from https://github.com/epoberezkin/fast-deep-equal/blob/master/src/index.jst
```

Comment:
Unfortunately we can't include code from a 3p like this (with a special exception to Preact code, which has given us permission). I think all that's needed for our use case is a shallow equals? If so, we could take advantage of `objectsEqualShallow` in `src/core/types/object/index.js`

Reply:
I replaced it with a simpler implementation.  I would've liked adding `fast-deep-equal` as a dependency, but for tiny functions like this, inline seems better.

Shallow won't work for the use-case of:
```
useMutationObserver(elRef, { attributeFilters: [ 'expanded' ] }, ...);
```